stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java

```diff
@@ -260,6 +260,8 @@ public void run() {
         LogUtil.w("I/O error: %s", ex);
       } catch (HttpException ex) {
         LogUtil.w("Unrecoverable HTTP protocol violation: %s", ex);
+      } catch (OutOfMemoryError ex) {
```

Comment:
True, but I noticed that this is not always the case.
For example, to work on this, I was using the Stetho sample:

- Use Picasso to download an oversized image (I used http://upload.wikimedia.org/wikipedia/commons/2/24/Willaerts_Adam_The_Embarkation_of_the_Elector_Palantine_Oil_Canvas-huge.jpg), so it won't fail due to the image size in the ImageView, only for the network request.
- Use OkHttpClient with network interceptors to intercept Picasso's network request.

This is the code in `APODPostsAdapter#bindView`
```java
      OkHttpClient okclient = new OkHttpClient();
      okclient.networkInterceptors().add(new StethoInterceptor());

      Picasso picasso = new Picasso.Builder(APODActivity.this)
        .downloader(new OkHttpDownloader(okclient))
        .build();

      picasso.load("http://upload.wikimedia.org/wikipedia/commons/2/24/Willaerts_Adam_The_Embarkation_of_the_Elector_Palantine_Oil_Canvas-huge.jpg?" + cursor.getPosition())
          .fit()
          .centerCrop()
          .into(holder.image);
```

After doing that, this is the stack trace:
```
03-15 11:11:59.140    6755-7032/com.facebook.stetho.sample E/AndroidRuntime﹕ FATAL EXCEPTION: StethoWorker12
    Process: com.facebook.stetho.sample, PID: 6755
    java.lang.OutOfMemoryError
            at java.lang.AbstractStringBuilder.enlargeBuffer(AbstractStringBuilder.java:94)
            at java.lang.AbstractStringBuilder.append0(AbstractStringBuilder.java:132)
            at java.lang.StringBuilder.append(StringBuilder.java:124)
            at org.json.JSONStringer.string(JSONStringer.java:344)
            at org.json.JSONStringer.value(JSONStringer.java:252)
            at org.json.JSONObject.writeTo(JSONObject.java:672)
            at org.json.JSONStringer.value(JSONStringer.java:237)
            at org.json.JSONObject.writeTo(JSONObject.java:672)
            at org.json.JSONObject.toString(JSONObject.java:641)
            at com.facebook.stetho.inspector.ChromeDevtoolsServer.handleRemoteRequest(ChromeDevtoolsServer.java:135)
            at com.facebook.stetho.inspector.ChromeDevtoolsServer.handleRemoteMessage(ChromeDevtoolsServer.java:104)
            at com.facebook.stetho.inspector.ChromeDevtoolsServer.onMessage(ChromeDevtoolsServer.java:80)
            at com.facebook.stetho.websocket.WebSocketSession$1.handleTextFrame(WebSocketSession.java:169)
            at com.facebook.stetho.websocket.WebSocketSession$1.onCompleteFrame(WebSocketSession.java:129)
            at com.facebook.stetho.websocket.ReadHandler.readLoop(ReadHandler.java:39)
            at com.facebook.stetho.websocket.WebSocketSession.handle(WebSocketSession.java:38)
            at com.facebook.stetho.websocket.WebSocketHandler.doUpgrade(WebSocketHandler.java:147)
            at com.facebook.stetho.websocket.WebSocketHandler.handleSecured(WebSocketHandler.java:99)
            at com.facebook.stetho.server.SecureHttpRequestHandler.handle(SecureHttpRequestHandler.java:45)
            at org.apache.http.protocol.HttpService.doService(HttpService.java:243)
            at org.apache.http.protocol.HttpService.handleRequest(HttpService.java:187)
            at com.facebook.stetho.server.LocalSocketHttpServer$WorkerThread.run(LocalSocketHttpServer.java:255)
```

So maybe in `ChromeDevtoolsServer#handleRemoteRequest`?